### ISSUE TITLE
Default plot options for log priors

### ISSUE THREAD

**Issue by ben18785** [pints-team/pints#741]
I constantly need to check the shape of priors that I am using and so need to do:

- values = np.linspace(min, max, 1000)
- y = map(lambda x: log_prior(x), values)
- plt.plot(values, y)
- plt.show()

Would anyone mind if we had a convenience function that did this for each prior? It could work by specifying log_prior.plot() where we pick default min and maxes for each univariate density, or optionally allow a user to pass these.

**Comment by MichaelClerx** [comment]
We did decide at one point not to have plotting functionality outside of `pints.plot`, so it'd be best if we could make it work with the interface:
```
import pints
import pints.plot

pints.plot.logprior(pints.GaussianLogPrior)
```

**Comment by fcooper8472** [comment]
@MichaelClerx just thinking about how that would work...

```python
pints.plot.logprior(pints.GaussianLogPrior)
```
Then surely `pints.plot.logprior` would have to be specialised for every log prior, as (e.g.) want to plot a Gaussian some distance either side of the mean but a Beta just between 0 and 1.

**Comment by MichaelClerx** [comment]
Yeah, either that or add some optional method which returns a range for a prior to be plotted in? This is closer to Ben's proposal, but avoids having plot commands in the priors... I'm not sure how I feel about this whole thing!

**Comment by fcooper8472** [comment]
See [here](https://stackoverflow.com/questions/1247486/list-comprehension-vs-map) for a discussion.

The main issue here was that (for some reason - not sure why!) the `map` version wasn't working on some Python versions, whereas the list comprehension seems OK everywhere.

**Comment by MichaelClerx** [comment]
Is this still relevant @ben18785 ?
